nil pointer dereference in `argocd app diff --local`

```
WARN[0037] Failed to remarshal oject: no kind "Issuer" is registered for version "certmanager.k8s.io/v1alpha1" in scheme "k8s.io/kubernetes/pkg/kubectl/scheme/scheme.go:28"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b11841]

goroutine 1 [running]:
github.com/argoproj/argo-cd/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetLabels(0x0, 0x0)
        /Users/jsuen/go/src/github.com/argoproj/argo-cd/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go:365 +0x41
github.com/argoproj/argo-cd/util/kube.SetAppInstanceLabel(0x0, 0x7ffeefbff5a0, 0x7, 0x314ac40, 0xc4204d6300)
        /Users/jsuen/go/src/github.com/argoproj/argo-cd/util/kube/kube.go:167 +0x7a
github.com/argoproj/argo-cd/cmd/argocd/commands.NewApplicationDiffCommand.func1(0xc420539d40, 0xc4203b07e0, 0x1, 0x3)
        /Users/jsuen/go/src/github.com/argoproj/argo-cd/cmd/argocd/commands/app.go:682 +0x101f
github.com/argoproj/argo-cd/vendor/github.com/spf13/cobra.(*Command).execute(0xc420539d40, 0xc4203b0720, 0x3, 0x3, 0xc420539d40, 0xc4203b0720)
        /Users/jsuen/go/src/github.com/argoproj/argo-cd/vendor/github.com/spf13/cobra/command.go:702 +0x2c6
github.com/argoproj/argo-cd/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4202ab200, 0xc4203afb00, 0xc4202a0480, 0xc4204190a0)
        /Users/jsuen/go/src/github.com/argoproj/argo-cd/vendor/github.com/spf13/cobra/command.go:783 +0x2e4
github.com/argoproj/argo-cd/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4202ab200, 0xc4204bbf78, 0xc4200b8058)
        /Users/jsuen/go/src/github.com/argoproj/argo-cd/vendor/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
        /Users/jsuen/go/src/github.com/argoproj/argo-cd/cmd/argocd/main.go:14 +0x27
```

--- a/cmd/argocd/commands/app.go
+++ b/cmd/argocd/commands/app.go
@@ -679,8 +679,11 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 
 					key := kube.NewResourceKey(res.Group, res.Kind, res.Namespace, res.Name)
 					if local, ok := localObjs[key]; ok || live != nil {
-						err = kube.SetAppInstanceLabel(local, appName)
-						errors.CheckError(err)
+						if local != nil {
+							err = kube.SetAppInstanceLabel(local, appName)
+							errors.CheckError(err)
+						}
+
 						items = append(items, struct {
 							key    kube.ResourceKey
 							live   *unstructured.Unstructured
@@ -732,7 +735,7 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 				// Diff is already available in ResourceDiff Diff field but we have to recalculate diff again due to https://github.com/yudai/gojsondiff/issues/31
 				diffRes := diff.Diff(item.target, item.live)
 				fmt.Printf("===== %s/%s %s/%s======\n", item.key.Group, item.key.Kind, item.key.Namespace, item.key.Name)
-				if diffRes.Modified {
+				if diffRes.Modified || item.target == nil || item.live == nil {
 					formatOpts := formatter.AsciiFormatterConfig{
 						Coloring: terminal.IsTerminal(int(os.Stdout.Fd())),
 					}

